### PR TITLE
Vts fixes for Android 10

### PR DESCRIPTION
**Pull request by arminn**

### CODE REVIEW

File: VisDataProvider.cpp [hunk 1]
```diff
@@ -179,6 +179,17 @@ int VisDataProvider::pull() {
     return 0;
 }
 
+bool VisDataProvider::waitConnection(int s) const {
+    ALOGV("waitConnectionMs");
```
Comment (rshym) [inline VisDataProvider.cpp:183]: Maybe it's better to log same name as function name, or log message with "%d ms".

File: VisDataProvider.cpp [hunk 2]
```diff
@@ -179,6 +179,17 @@ int VisDataProvider::pull() {
...
+    int minSleepTime = std::min(s, 1);
```
Comment (rshym) [inline VisDataProvider.cpp:184]: As I understand this line is equal to "int minSleepTime = 1;".
std::min will return 1 always except "s=0", but this case I will review bellow.

File: VisDataProvider.cpp [hunk 3]
```diff
@@ -179,6 +179,17 @@ int VisDataProvider::pull() {
...
+    int timeSleeped = 0;
+    while (mConnectedState != STATE_CONNECTED) {
+        sleep(minSleepTime);
+        timeSleeped += minSleepTime;
+        if (timeSleeped >= s) break;
+    }
```
Comment (rshym) [inline VisDataProvider.cpp:190]: As I understand, if (s=0) we will:
{
  sleep(0);
  0+=0;
  if (0>=0) break;
}
As result we will return STATE_CONNECTED instantly in any case.
Maybe it's better to implement this function as "check for connection without real sleep if (s==0)".

Comment (arminn) [reply VisDataProvider.cpp:190]: With sleep time 0 it is logically correct, that client doesn't want to wait. 
In any case, mConnectedState == STATE_CONNECTED will be returned, but not STATE_CONNECTED.

Comment (rshym) [reply VisDataProvider.cpp:190]: My bad. I confused '=' and '=='.
Please consider initial comment as invalid.

**Review by rshym — APPROVED** [review]

**Comment by rshym** [comment]
LGTM

**Pull request merged**